web/src/compose_paste.ts

```diff
@@ -158,13 +158,39 @@ function is_from_excel(html_fragment: HTMLBodyElement): boolean {
     return true;
 }
 
+function is_from_libreoffice_calc(html_fragment: HTMLBodyElement): boolean {
+    const html_tag = html_fragment.parentElement;
+    if (!html_tag || html_tag.nodeName !== "HTML") {
+        return false;
+    }
+
+    const has_libreoffice_metadata = [...html_tag.querySelectorAll("meta")].some(
+        (meta) => meta.name === "generator" && meta.content.startsWith("LibreOffice"),
+    );
+    if (!has_libreoffice_metadata) {
+        return false;
+    }
+
+    const body_tag = html_tag.querySelector("body");
```

Comment:
We pass in the body element to the function. We could just use the `html_fragment` passed in?